src/components/LoginForm/LoginForm.tsx

```diff
@@ -0,0 +1,121 @@
+import React, {
+  useState,
+  useContext,
+  MouseEvent,
+  KeyboardEvent,
+} from 'react';
+import classNames from 'classnames';
+
+import { Context } from '../context';
+
+import { getUsers, addUsers } from '../../api';
+
+export const LoginForm: React.FC = () => {
+  const {
+    setUser,
+    setChangeUserButton,
+    error,
+  } = useContext(Context);
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState(false);
+  const [emailLoader, setEmailLoader] = useState(false);
+
+  // eslint-disable-next-line max-len
+  const regexp = /^((([0-9A-Za-z]{1}[-0-9A-z\\.]{0,30}[0-9A-Za-z]?)|([0-9А-Яа-я]{1}[-0-9А-я\\.]{0,30}[0-9А-Яа-я]?))@([-A-Za-z]{1,}\.){1,}[-A-Za-z]{2,})$/g;
+
+  // eslint-disable-next-line max-len
+  const loginEvent = (e: MouseEvent<HTMLButtonElement> | KeyboardEvent<HTMLInputElement>) => {
+    e.preventDefault();
+
+    if (!email.match(regexp)) {
+      setEmailError(true);
+
+      return;
+    }
+
+    setEmailLoader(true);
+
+    getUsers(email)
+      .then((user) => {
+        if (user.length !== 0) {
+          setUser(user[0]);
+          setEmailLoader(false);
+          localStorage.setItem('user', JSON.stringify(user[0]));
+          window.location.hash = '/todos';
+        } else {
+          addUsers(email)
+            .then(() => {
+              setUser(user[0]);
+              setEmailLoader(false);
+              localStorage
+                .setItem('user', JSON.stringify(user[0]));
+            }).catch(() => {
+              error('Server error');
+              setEmailLoader(false);
```

Comment:
this code hard to read - try to rewrite, or make separate functions and call them here